CMakeLists.txt

```diff
@@ -87,14 +87,17 @@ option(USE_VALGRIND "Use valgrind when testing" true)
 
 option(WITH_RUST "Use rust implemented internals" true)
 if(WITH_RUST)
-    include(FetchContent)
-    FetchContent_Declare(
-        Corrosion
-        GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
-        GIT_TAG be76480232216a64f65e3b1d9794d68cbac6c690 # v0.4.5
-        FIND_PACKAGE_ARGS
```

Comment:
With new enough CMake (3.24.0), this flag tells CMake to use system packages first.

https://cmake.org/cmake/help/latest/module/FetchContent.html#integrating-with-find-package

I'd like to investigate why it didn't work as intended before merge this change.

Reply:
yeah, totally, trying to file the patch so that I can unblock the homebrew pr. (we have policies to disable `FetchContent_Declare` usage now)

Reply:
It looks like brew unconditionally trapped FetchContent

https://github.com/Homebrew/brew/pull/17310

Reply:
yeah, that is what I referred to, it is good though (we do prefer system dependency)